src/couch/src/couch_doc.erl

```diff
@@ -154,7 +155,21 @@ from_json_obj_validate(EJson) ->
     from_json_obj_validate(EJson, undefined).
 
 from_json_obj_validate(EJson, DbName) ->
-    MaxSize = config:get_integer("couchdb", "max_document_size", 4294967296),
+    ConfigMaxSize = config:get_integer(
+        "couchdb",
+        "max_document_size",
+        ?DOCUMENT_SIZE_LIMIT_BYTES
+    ),
+    case ConfigMaxSize > ?DOCUMENT_SIZE_LIMIT_BYTES of
+        true ->
+            couch_log:error(
```

Comment:
I am wondering if there is a better way to handle logging. One one hand, we want to inform the user their limit value was overridden, but on the other hand, this would flood the logs on every single doc operation whether doc size fits or doesn't fit the under the limit.

One idea is perhaps we could check these hard-coded limits once in `fabric2_server` or other singleton process on start, emit the warning to the logs and then don't log it after that. 

@iilyak @noahshaw11 what do you think?

Reply:
I have moved the warnings to `fabric2_server` in `init(_)`. This ensures that the limits are only logged once. See https://github.com/apache/couchdb/pull/3632/commits/468123da973d22ff9d4e91477b1f48568cde80b9.